numba/tests/test_svml.py

```diff
@@ -6,15 +6,172 @@
+       body = '%s%s[0] += math.%s(%s)\n'%(pad, res, func, arg_list)
+    elif mode == "numpy":
+       body = '%s%s += np.%s(%s)\n'%(pad, res, func, ','.join(args))
+    else:
+       assert mode == "range" or mode == "prange"
+       arg_list = ','.join([a+'[i]' for a in args])
+       body = '{pad}for i in {mode}({res}.size):\n{pad}{pad}{res}[i] += math.{func}({arg_list})\n'.format(**locals())
+    # TODO: refactor that for-loop goes into umbrella function, 'mode' can be 'numpy', '0', 'i' instead
+    # TODO: that will enable mixed usecases like prange + numpy
+    scalar_func = '$_'+func if numba.config.IS_OSX else '$'+func
+    if mode == "scalar":
+       avoids   = ['__svml']
+       contains = [scalar_func]
+    else: # will vectorize
+       contains = ['__svml_%s%d%s,'%(func, vlen, '' if getattr(flags, 'fastmath', False) else '_ha')]
+       avoids   = []  # TODO: [scalar_func] - if possible, force LLVM to prevent generating the failsafe scalar paths
+    return body, contains, avoids
+
+
+def usecase_name(dtype, mode, vlen, flags):
+    """ Returns pretty name for given set of modes """
+
+    return "{dtype}_{mode}{vlen}_{flags.__name__}".format(**locals())
+
+
+def combo_svml_usecase(dtype, mode, vlen, flags):
+    """ Combine multiple function calls under single umbrella usecase """
+
+    name = usecase_name(dtype, mode, vlen, flags)
+    body = """def {name}(n):
+        ret = np.empty(n*8, dtype=np.{dtype})
+        x   = np.empty(n*8, dtype=np.{dtype})\n""".format(**locals())
+    funcs = numpy_funcs if mode == "numpy" else other_funcs
+    contains = []
+    avoids = []
+    for f in funcs:
+        b, c, a = func_patterns(f, ['x'], 'ret', mode, vlen, flags)
+        avoids += a
+        body += b
+        contains += c
+    body += " "*8 + "return ret"
+    ldict = {}
+    exec_(body, globals(), ldict)
+    ldict[name].__doc__ = body
+    return ldict[name], contains, avoids
+
+
+@needs_svml
+class TestSVMLGeneration(TestCase):
+    """ Tests all SVML-generating functions produce desired calls """
+
+    # env mutating, must not run in parallel
+    _numba_parallel_test_ = False
+
+    @classmethod
+    def _inject_test(cls, dtype, mode, vlen, flags):
+        args = (dtype, mode, vlen, flags)
+        def test_template(self):
+            fn, contains, avoids = combo_svml_usecase(*args)
+            # look for specific patters in the asm for a given target
+            with override_env_config('NUMBA_CPU_NAME', vlen2cpu[vlen]), \
+                 override_env_config('NUMBA_CPU_FEATURES', ''):
+                # recompile for overridden CPU
+                jit = compile_isolated(fn, (numba.int64,), flags=flags)
+            asm = jit.library.get_asm_str()
+            missed = [pattern for pattern in contains if not pattern in asm]
+            found  = [pattern for pattern in avoids if pattern in asm]
+            assert not missed and not found, "While expecting %s,\n\tit contains %s:\n%s\nwhen compiling %s" % \
```

Comment:
Perhaps use members of the `self.assert*` family to be consistent with other test code.